lib/asynchronize.rb

```diff
@@ -1,14 +1,19 @@
 module Asynchronize
   require 'set'
   def self.included(base)
+      
+    # @@asynced_methods     ... methods we have already asynchronized.
+    # @@methods_to_async    ... methods that should be asynchronized.
+    # @@methods_asyncing    ... what is the short version of this?
```

Comment:
This is just a guard for method_added, so it can keep track of which method(s) it's modifying.
Otherwise, it goes into infinite recursion trying to redefine the method it's redefining.
A short version could maybe be 'methods we are currently processing'

Reply:
Also, these should probably go underneath base.class_eval, since they apply to the class and not the module